### ISSUE TITLE
Error while deleting an account

### ISSUE THREAD

**Issue by vmario89** [nuxsmin/sysPass#810]
Dear developers,

we are currently exploring the feature set of sysPass. So i created some test accounts and played with data.

Now i have the problem that i cannot delete an account. I am logged is as admin and other accounts could be deleted without problems. I tried it many times, but same error in event log all the time:

  | Error while queryingSQLSTATE[23000]: Integrity  constraint violation: 1451 Cannot delete or update a parent row: a  foreign key constraint fails (`syspass`.`accFiles`,CONSTRAINT `fk_accFiles_accounts_id` FOREIGN KEY (`accfile_accountId`) REFERENCES `accounts` (`account_id`) ON DELETE NO ACTION ON UPDATE NO ACTION) (0)SQL : 'DELETE FROM accounts WHERE account_id = ? LIMIT 1'
-- | --

may this account cannot be deleted because it has a shared link which also does not work. i attached some screenshots.

the ressource view:
![grafik](https://user-images.githubusercontent.com/5631071/34381318-183d5f46-eb08-11e7-9ab1-89ad75de2b92.png)

the edit view:
![grafik](https://user-images.githubusercontent.com/5631071/34381303-048e4a82-eb08-11e7-8ce0-f87bca1f66b3.png)

the share link, which also does not work (no access, but i am admin):
![grafik](https://user-images.githubusercontent.com/5631071/34381310-0dc1fd4c-eb08-11e7-956f-21a1823faafa.png)

the failure messege in event log
![grafik](https://user-images.githubusercontent.com/5631071/34381300-fa488f4c-eb07-11e7-80db-2e067dcb0b8c.png)

**Comment by vmario89** [comment]
found the solution: the account got attached file. after removing the file the account could be deleted. May a hint should be given that deletion will fail if attachments are present.

**Comment by nuxsmin** [comment]
@vmario89 you're right, a hint should be displayed about deleting a linked file. Noted for the next release ;)

BTW, be aware that almost everything in the database is related, since it avoids the duplicity of the data and increases the performance.

Thanks for the feedback!

**Comment by vmario89** [comment]
Hello @nuxsmin i checked many more features provided by sysPass. In general i recognized that it's not really easy to delete some data. Please let me explain my fresh user experience with this linkage concept/duplicity (which is intionally really good!).

I tell you my use case: I installed sysPass on fresh LAMP server and then i imported a KeePass xml file with ~25 accounts. sysPass converted these accounts into sysPass accounts and provided some automatically generated data (KeePass data folders converted to sysPass categories, ..). 

Because i redesigned the KeePass data in sysPass now by hand, renamed categories. I also wanted to delete categories and also some accounts, which were not longer in use now. the "problem" is: 
now unused categories cannot be deleted until i delete all items in "Accounts (H)" which contains the deleted accounts. So there is invisible linkage between deleted accounts and present categories until deletion of the Accounts (H) history.

The following problem of the problem is: in Accounts (H) i cannot see the items, which contain the category, which i cannot delete. The list screen only shows the account and customer. So my only solution is to clear complete history to delete my unused categories.

the question of the problem of the problem: what's the reason for existence of Accounts (H) history? 
For auditing review i can use the "Event Log". For checking Account change history i can use the history function at the account itself.

At first i thought this Accounts (H) feature could work like a trash can (this is also an idea): If a user deletes an account it could be dropped into this tray. Feature could be to bring back some account data by user/admin from trash.

hope you can follow my thoughts. Sometimes it's hard to describe. I will post some screenshots later when i am back at work. I think this will help to better undesrtand what i am talking about.

best regards, Mario :)

**Comment by nuxsmin** [comment]
Hello @vmario89

I see your points, and I think that linked items can be a hassle many times. In this use case, the accounts history could be undone by selecting a restore point through the account details view, so we need to think what's more valuable, either history integrity or deleting a category.

It would be an option to ask for related items when deleting a linked item, so the user could decide which is the best way.

Thanks for the feedback!

Kind regards

**Comment by vmario89** [comment]
asking the user for deleting related items sounds good. for a "quick bugfix" it just would help to give more detail in error/warning popups for the user to inform about the data struggles so the user/admin can react on this in a proper way. i think there also should be shown some more item information in list views in Accounts (H) about the invisible relations.

**Comment by dajones70** [comment]
This is also true when renaming an account.  It can't be deleted.  Also, a renamed account can't login.  I suspect that the master password is being used along with the username to determine/verify the login authentication.  When I renamed the account back to the original name, I was able to login again after it prompted me for the previous password which was the same as the current password.
